text/XXXX-replace_publish_clone_with_push_pull.md

```diff
@@ -0,0 +1,258 @@
+
+If the dataset you are checking out isn't in the qri database, the latest 
+version of the dataset will be pulled from the network first. To prevent network
+requests, use the --offline flag.
+
+Usage:
+  qri checkout DATASET [flags]
+
+Examples:
+  # Place a copy of me/annual_pop in the ./annual_pop directory:
+  $ qri checkout me/annual_pop
+
+Flags:
+  -h, --help      help for checkout
+      --offline   don't fetch datasets from the network
+
+Global Flags:
+      --ipfs-path string   override IPFS path location (default "/Users/b5/.ipfs")
+      --log-all            log all activity
+      --no-color           disable colorized output
+      --no-prompt          disable all interactive prompts
+      --repo string        provide a path to load qri from (default "/Users/b5/.qri")
+```
+
+### No more clone
+We already have "clone" language in qri cloud, and the original plan has been to rename `add` to `clone`. In discussion we've come to realize The problem is add isn't the same thing as clone when compared to git. Git needs `clone` because git does not have a built-in naming system. Clone copies a git repo _from a place_, usually a URL. 
+
+Because qri can resolve dataset names without additional information from the user we can add `pull` directly to `checkout`, and drop the term `clone` from our lexicon. We can use this saved term later for something like "following" a dataset. See the unsreolved questions section for more info.
+
+### Automatic Pulling & Networking UX
+Giving `checkout` the power to pull `pull` when data isn't local is the first concrete example of _automatic pulling_. This change means checkout by default assumes it can use the network to fulfill requests, which is is a shift in our user experience. We've wrestled with weather qri should default to making network requests for users for some time. As an example `qri --help` has a set of "Network commands", implying the other commands in qri _don't_ use the network. This RFC proposes we break that distinction, and assume that any command within qri _might_ use the network, unless instructed otherwise with either a configuration value or an `--offline` flag.
+
+Automatic pulling requires us to make a decision on this, and I think it's the right choice is to assume network access by default. We can take a cue from package managers, which also have a namespace, and assume network acess, automatically making network requests when the user seeks datasets they don't have. `npm install some_package` downloads from the web. Both npm and qri have name resolution systems that make this process possible. From a UX perspective npm lets me _ignore_ the network and focus on writing software.
+
+We've struggled with the word "network" because unlike npm, qri also has a peer-2-peer system, but I think the simple answer to needing assets that may only be available in a p2p context are to show an error asking users to run `qri connect`.
```

Comment:
IIInnnnnteresting.
Haven't dug into this in a while, but are we anticipating most remotes will be accessed via HTTP? Or would pushing to a remote (a trussed up peer) be a p2p interaction? Both, depending on the multiaddress/uri that is given to talk to the remote?